go/contacts/contacts_test.go

```diff
@@ -453,3 +453,48 @@ func TestErrorsInResolution(t *testing.T) {
 	}
 	require.Equal(t, expected, displayResults(res))
 }
+
+func TestDuplicateEntries(t *testing.T) {
+	tc := libkb.SetupTest(t, "TestLookupContacts", 1)
+	defer tc.Cleanup()
+
+	contactList := []keybase1.Contact{
+		keybase1.Contact{
+			Name: "Alice",
+			Components: []keybase1.ContactComponent{
+				makePhoneComponent("home", "+1111222"),
+				makePhoneComponent("car", "+1111222"),
+				makePhoneComponent("car", "+1111222"),
+			},
+		},
+		keybase1.Contact{
+			Name: "Bob",
+			Components: []keybase1.ContactComponent{
+				makeEmailComponent("email", "bob+test@keyba.se"),
+			},
+		},
+		keybase1.Contact{
+			Name: "Robert B.",
+			Components: []keybase1.ContactComponent{
+				makeEmailComponent("E-Mail", "bob+test@keyba.se"),
+			},
+		},
```

Comment:
I think you need another `keybase1.Contact` whose name is also `Alice` and has the phone number `+1111222`. I'm pretty sure you don't handle that case right now, and that's another level of deduping that would be handy.